pkg/driver/node_server.go

```diff
@@ -256,6 +256,27 @@ func (driver *Driver) nodeStageVolume(
 		return nil // volume already staged, do nothing and return here
 	}
 
+	if volumeContext[volEncryptionSecretNameKey] != "" {
+		log.Infof("Requested volume needs encryption. Received Secret name: %s, Secret namespace: %s",
+			volumeContext[volEncryptionSecretNameKey], volumeContext[volEncryptionSecretNamespaceKey])
+
+		var encPassphrase string
+		encPassphrase, err = driver.getEncryptionPassphraseFromSecret(volumeContext[volEncryptionSecretNameKey],
+																	  volumeContext[volEncryptionSecretNamespaceKey])
+		if err != nil {
+			log.Errorf("Encountered error while fetching secret - %v", err.Error())
+			return err // specified secret for encryption key doesn't exist
+		}
+
+		ctxt := make(map[string]string)
+		for k,v := range publishContext {
+			ctxt[k] = v
+		}
+		ctxt[volEncryptionKey] = encPassphrase
+		publishContext = ctxt
+		log.Tracef("Modified publish context: %v", publishContext)
```

Comment:
Are we printing the encPassphrase here as part of publishContext?

Reply:
This was for debugging purpose. Will be removing it.